File: packages/wnode-sdk-ts/src/mesh/state.ts
import { SecuritySnapshot } from './health';
import { PeerInfo, WorkflowStepAssignment } from './types';
import { ProofOfCompute } from '../types';
import { IncidentRecord } from './byzantine';

export interface NodeState {
  nodeId: string;
  capabilities: string[];
  authStatus: 'TRUSTED' | 'SUSPICIOUS' | 'QUARANTINED';
  healthSnapshot: {
    activePeers: number;
    isHealthy: boolean;
  };
  securitySnapshot: SecuritySnapshot;
}

export interface TransportSnapshot {
  connectedPeers: string[];
  sequenceNumber: number;
}

export interface MeshState {
  version: string;
  timestamp: number;
  localNodeId: string;
  peerTable: Record<string, NodeState>;
  activeWorkflows: Record<string, WorkflowStepAssignment[]>;
  aggregatedProofs: Record<string, ProofOfCompute>;
  incidentLogs: Record<string, IncidentRecord[]>;
  transportSnapshot: TransportSnapshot;
}

File: archive/root_legacy/wnode-sdk-go/mesh/state.go
package mesh

type NodeState struct {
	NodeID           string
	Capabilities     []string
	AuthStatus       string
	HealthSnapshot   map[string]interface{}
	SecuritySnapshot SecuritySnapshot
}

type TransportSnapshot struct {
	ConnectedPeers []string
	SequenceNumber int
}

type MeshState struct {
	Version           string
	Timestamp         int64
	LocalNodeID       string
	PeerTable         map[string]NodeState
	ActiveWorkflows   map[string]interface{}
	AggregatedProofs  map[string]interface{}
	IncidentLogs      map[string][]IncidentRecord
	TransportSnapshot TransportSnapshot
}
